### PR TITLE
feat: ENT-7519 Added EnterpriseSubsidyBudget model

### PR DESCRIPTION
**Pull request by irfanuddinahmad**
**Description**: This PR creates the model and api for policy/budget aggregates - EnterpriseSubsidyBudget

**JIRA: [ENT-7559](https://2u-internal.atlassian.net/browse/ENT-7559)

**Output**:
```
{
    "offer_id": "72550789-6b03-4e58-94d0-1ec49c87db4a",
    "budgets": [
        {
            "id": 1,
            "subsidy_access_policy_uuid": "1fff3a0e-f957-43b3-b3e4-9e9c0f24be7b",
            "subsidy_uuid": "72550789-6b03-4e58-94d0-1ec49c87db4a",
            "enterprise_customer_uuid": "1386b06c-9669-435d-8f5c-ee757e642c3d",
            "enterprise_customer_name": "MAX Inc",
            "subsidy_access_policy_description": null,
            "subsidy_title": null,
            "catalog_name": null,
            "subsidy_access_policy_type": null,
            "date_created": null,
            "subsidy_start_datetime": null,
            "subsidy_expiration_datetime": null,
            "is_active": false,
            "is_test": false,
            "ocm_usage": null,
            "exec_ed_usage": null,
            "usage_total": null,
            "enterprise_contract_discount_percent": null,
            "starting_balance": null,
            "amount_of_policy_spent": null,
            "remaining_balance": null,
            "percent_of_policy_spent": null
        }
    ],
    "enterprise_customer_uuid": "1386b06c-9669-435d-8f5c-ee757e642c3d",
    "enterprise_name": "MAX Inc",
    "sum_amount_learner_paid": null,
    "sum_course_price": null,
    "status": null,
    "offer_type": null,
    "date_created": null,
    "emails_for_usage_alert": null,
    "discount_type": null,
    "discount_value": null,
    "max_discount": null,
    "total_discount_ecommerce": null,
    "amount_of_offer_spent": null,
    "percent_of_offer_spent": null,
    "remaining_balance": null,
    "amount_offer_spent_ocm": null,
    "amount_offer_spent_exec_ed": null
}
```

**Merge checklist:**
- [ ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production but edx-analytics-data-api doesn't install it
    - `test-master.in` if edx-analytics-data-api pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
- [ ] `make static` has been run to update webpack bundling if any static content was updated
- [ ] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the edx-analytics-data-api requirements installed or if you checked out edx-enterprise-data into the src directory used by edx-analytics-data-api, you can run this command through an edx-analytics-data-api shell.
        - It would be `./manage.py makemigrations` in the shell.
- [ ] [Version](https://github.com/openedx/edx-enterprise-data/blob/master/enterprise/__init__.py) bumped
- [ ] [Changelog](https://github.com/openedx/edx-enterprise-data/blob/master/CHANGELOG.rst) record added
- [ ] Translations updated (see docs/internationalization.rst but also this isn't blocking for merge atm)

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/openedx/edx-enterprise-data/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [Travis](https://travis-ci.org/github/edx/edx-enterprise-data), verify version has been pushed to [PyPI](https://pypi.org/project/edx-enterprise-data/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [edx-analytics-data-api](https://github.com/openedx/edx-analytics-data-api) to upgrade dependencies (including edx-enterprise-data)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-analytics-data-api will look for the latest version in PyPi.
    - Note: the edx-enterprise-data constraint in edx-analytics-data-api **must** also be bumped to the latest version in PyPi.

### CODE REVIEW

**Comment by adamstankiewicz** [comment]
[question/clarification] What is intended to create/populate the `EnterpriseSubsidyBudget` model with policy/budget metadata data?

File: enterprise_data/api/v1/serializers.py [hunk 1]
```diff
@@ -57,10 +58,30 @@ def get_total_learning_time_hours(self, obj):
...
 class EnterpriseOfferSerializer(serializers.ModelSerializer):
     """
     Serializer for EnterpriseOfferSerializer model.
     """
+    budgets = serializers.SerializerMethodField()
+
+    def get_budgets(self, instance):
+        """Gets related budgets"""
+        if len(instance.offer_id) == 32:
+            budget_queryset = EnterpriseSubsidyBudget.objects.filter(subsidy_uuid=instance.offer_id)
```
Comment (muhammad-ammar) [inline enterprise_data/api/v1/serializers.py]: Should we first filter by `enterprise_customer_uuid` and then by `subsidy_uuid`?

Comment (irfanuddinahmad) [reply enterprise_data/api/v1/serializers.py]: @muhammad-ammar subsidy_uuid should return only a few rows ... filtering by enterprise_customer_uuid is not needed

**Comment by irfanuddinahmad** [comment]
> [question/clarification] What is intended to create/populate the `EnterpriseSubsidyBudget` model with policy/budget metadata data?

@adamstankiewicz correct

**Review by muhammad-ammar — APPROVED** [review]
LGTM

**Comment by adamstankiewicz** [comment]
> > [question/clarification] What is intended to create/populate the `EnterpriseSubsidyBudget` model with policy/budget metadata data?
> 
> @adamstankiewicz correct

@irfanuddinahmad My question here was more about what mechanism(s) we have to actually create/update instances of the `EnterpriseSubsidyBudget` model, as this PR only created the model and returned it in the API response. It doesn't answer the question of how these model instances get created in the first place at first glance?

**Pull request merged**